src/text/Word2Vec.ts

```diff
@@ -0,0 +1,126 @@
+import * as tf from '@tensorflow/tfjs';
+import { createData, createUniqueWord, objectLength, skipGram, UniqueWord } from './word2vec-utils';
+
+export default class Word2Vec {
+  public weight: number[][] | undefined = undefined;
+  public uniqueWords: UniqueWord;
+  private data: number[][];
+  private label: number[];
+  private nwords: number;
+  private size: number;
+
+  /**
+   *
+   * @param text Array of strings
+   * @param window window size for skip-gram
+   * @param stopwords
+   * @param size embedding dimension
+   */
+  constructor(text: string[], window: number, stopwords: string[] | undefined, size = 50) {
+    const [ wordList, allText ] = skipGram(window, text, stopwords);
+    this.uniqueWords = createUniqueWord(allText as string[]);
+    this.nwords = objectLength(this.uniqueWords);
+    const [ data, label ] = createData(wordList as string[][], this.uniqueWords, this.nwords);
+    this.data = data as number[][];
+    this.label = label as number[];
+    this.size = size;
+  }
+
+  /**
+   * Train word2vec model
+   * @param epochs
+   * @param verbose to print out the training logs or not
+   * @return Promise
+   */
+  public async train(epochs = 100, verbose = 0): Promise<void> {
+    const model = tf.sequential();
+    model.add(tf.layers.dense({ units: this.size, inputShape: [ this.nwords ] }));
+    model.add(tf.layers.dense({ units: this.nwords, activation: 'softmax' }));
+    model.compile({ optimizer: 'sgd', loss: 'sparseCategoricalCrossentropy', metrics: [ 'accuracy' ] });
+
+    const dataTensor = tf.tensor(this.data);
+    const labelTensor = tf.tensor(this.label);
+
+    await model.fit(dataTensor, labelTensor, {
+      epochs: epochs,
+      callbacks: {
+        onEpochEnd: (epoch, log) => {
+          if (verbose){
+            console.log(`Epoch ${epoch}: loss = ${log.loss}`);
+          }
+        }
+      }
+    });
+    this.weight = model.getWeights()[0].arraySync() as number[][];
+  }
+
+  /**
+   *Calculate the similarity between two words.
+   * @param wordA
+   * @param wordB
+   * @return number
```

Comment:
How about writing the number range for the returned value?

Reply:
yeah. that will be cool